lib/policies/systemErrorRetryPolicy.ts

```diff
@@ -104,30 +105,38 @@ function updateRetryData(policy: SystemErrorRetryPolicy, retryData?: RetryData,
   // Adjust retry interval
   let incrementDelta = Math.pow(2, retryData.retryCount) - 1;
   const boundedRandDelta = policy.retryInterval * 0.8 +
-    Math.floor(Math.random() * (policy.retryInterval * 1.2 - policy.retryInterval * 0.8));
+    Math.floor(Math.random() * (policy.retryInterval * 0.4));
   incrementDelta *= boundedRandDelta;
 
   retryData.retryInterval = Math.min(policy.minRetryInterval + incrementDelta, policy.maxRetryInterval);
 
   return retryData;
 }
 
-function retry(policy: SystemErrorRetryPolicy, request: WebResourceLike, operationResponse: HttpOperationResponse, retryData?: RetryData, err?: RetryError): Promise<HttpOperationResponse> {
+async function retry(
+  policy: SystemErrorRetryPolicy,
+  request: WebResourceLike,
+  operationResponse: HttpOperationResponse,
+  err?: RetryError,
+  retryData?: RetryData
+): Promise<HttpOperationResponse> {
   retryData = updateRetryData(policy, retryData, err);
   if (err && err.code && shouldRetry(policy, retryData) &&
     (err.code === "ETIMEDOUT" || err.code === "ESOCKETTIMEDOUT" || err.code === "ECONNREFUSED" ||
       err.code === "ECONNRESET" || err.code === "ENOENT")) {
     // If previous operation ended with an error and the policy allows a retry, do that
-    return utils.delay(retryData.retryInterval)
-        .then(() => policy._nextPolicy.sendRequest(request.clone()))
-        .then(res => retry(policy, request, res, retryData, err))
-        .catch(err => retry(policy, request, operationResponse, retryData, err));
+    try {
+      await utils.delay(retryData.retryInterval);
+      return policy._nextPolicy.sendRequest(request.clone());
+    }
+    catch (err) {
```

Comment:
can we rename this exception error variable? it shadows the `err` in the function params and makes this more confusing than it needs to be